include/mbedtls/ssl_internal.h

```diff
@@ -158,17 +173,43 @@
 #error Bad configuration - record content too large.
 #endif
 
-#if MBEDTLS_SSL_PAYLOAD_LEN > 16384 + 2048
-#error Bad configuration - protected record payload too large.
+#if MBEDTLS_SSL_IN_CONTENT_LEN > MBEDTLS_SSL_MAX_CONTENT_LEN
+#error Bad configuration - incoming record content should not be larger than MBEDTLS_SSL_MAX_CONTENT_LEN.
```

Comment:
Not a blocker since it's preexisting, but since `check_config.h` exists, I think all configuration checks should be in `check_config.h`.